File: lab2c.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
void insertionsort(int *A,int n)
{
    int temp,j;
    for(int i=1;i<n;i++)
    {
        temp=A[i];
        for(j=i-1;j>-1&&temp<A[j];j--)
        {
            A[j+1]=A[j];
        }
        A[j+1]=temp;
    }
}
void generateRandomArray(int *arr,int n)
{
    for(int i=0;i<n;i++)
    {
        arr[i]=rand()%n;
    }
}
int main()
{
    int n;
    printf("Enter number of elements (n): ");
    scanf("%d", &n);
    int *arr=(int *)malloc(n*sizeof(int));;
    if(arr==NULL)
    {
        printf("Memory allocation fail");
    }
    generateRandomArray(arr,n);
    clock_t start=clock();
    for(int i=0;i<1000;i++)
    {
        insertionsort(arr,n);
    }
    clock_t end= clock();
    double time_taken=((double)(end-start))/CLOCKS_PER_SEC/1000.0;
    printf("Time taken to sort %d element : %f seconds\n",n,time_taken);
    free(arr);
    return 0;
}

File: lab2c.py
import matplotlib.pyplot as plt
n_values = [100, 500, 1000, 5000, 10000, 50000,100000]
time_values=[0.000001,0.000005,0.000010, 0.000034, 0.000111, 0.001808,0.006747] # in seconds
plt.plot(n_values, time_values, marker='o')
plt.title("Insertion Sort: Time vs Number of Elements")
plt.xlabel("Number of Elements (n)")
plt.ylabel("Time Taken (seconds)")
plt.grid(True)
plt.show()